### ISSUE TITLE
cloneInstance does not copy over modules and logs a warning because of it

### ISSUE THREAD

**Issue by dietergeerts** [i18next/i18next#1413]
I don't know what the actual bug is, the fact that the `modules` property is not copied over when using `cloneInstance`, or that the warning being logged isn't checking the correct things:

```
      if (!this.modules.languageDetector && !this.options.lng) {
        this.logger.warn('init: no languageDetector is used and no lng is defined');
      } // append api
```

This gives problems in my setup. I use https://github.com/i18next/i18next-express-middleware, and that will do a `cloneInstance` for each request. The language detector is being copied from `services`, but not from `modules`.

So if someone can enlighten me on this, that will be very much appreciated.

Reason for asking this: Currently, I'm having an actual bug, where the missing keys are not being saved. I'm trying to debug, and I see so many warning being logged, telling me there is something wrong with the config, but I just followed the guidelines in the readme's. This is making it difficult to find out what is going wrong.

**Comment by jamuhl** [comment]
feel free to provide a PR changing that to check for  `!this.services.languageDetector`

**Comment by adrai** [comment]
should be fixed with v19.3.4

**Comment by adrai** [comment]
If you like this module don’t forget to star this repo. Make a tweet, share the word or have a look at our https://locize.com to support the devs of this project.

If you liked my support / work - I would enjoy a coffee sponsored using the “💜Sponsor Button”.

There are many ways to help this project 🙏
